visualization/app/codeCharta/ui/attributeTypeSelector/attributeTypeSelector.component.ts

```diff
@@ -0,0 +1,67 @@
+import "./attributeTypeSelector.component.scss"
+import { StoreService } from "../../state/store.service"
+import { AttributeTypeValue } from "../../codeCharta.model"
+import { setAttributeTypes } from "../../state/store/fileSettings/attributeTypes/attributeTypes.actions"
+import { MetricService } from "../../state/metric.service"
+import { EdgeMetricDataService } from "../../state/edgeMetricData.service"
+
+export class AttributeTypeSelectorController {
+	private _viewModel: {
+		aggregationSymbol: string
+	} = {
+		aggregationSymbol: ""
+	}
+
+	private type: string
+	private metric: string
+
+	/* @ngInject */
+	constructor(
+		private storeService: StoreService,
+		private metricService: MetricService,
+		private edgeMetricDataService: EdgeMetricDataService
+	) {}
+
+	public $onInit() {
+		this.setAggregationSymbol()
+	}
+
+	public setToAbsolute(metricName: string, category: string) {
+		this.setAttributeType(metricName, category, AttributeTypeValue.absolute)
+	}
+
+	public setToRelative(metricName: string, category: string) {
+		this.setAttributeType(metricName, category, AttributeTypeValue.relative)
+	}
+
+	private setAttributeType(metricName: string, category: string, type: AttributeTypeValue) {
+		const attributeTypes = this.storeService.getState().fileSettings.attributeTypes
+		if (attributeTypes[category]) {
+			attributeTypes[category][metricName] = type
+		}
+		this.storeService.dispatch(setAttributeTypes(attributeTypes))
```

Comment:
This conflicts with the immutable update pattern of redux. You're copying a reference of the attributeTypes and changing the store without dispatching an action (even though you do it later).

Instead define another action in `attributeType.actions.ts` that you can dispatch in order to change an attributeType by category and metricName.

`setAttributeTypeByName(category, metricName)` and then `this.storeService.dispatch(setAttributeTypeByName(category, metricName)`

Reply:
You are right, I hence created a new aciton for this. Not sure if the updateAttributeType function is in the right place within the reducer, if not pls move it to  a better place.